src/providers/FusionAuthProvider.tsx

```diff
@@ -1,5 +1,108 @@
-import React from 'react';
+import React, { useCallback, useContext, useMemo } from 'react';
+import { TextEncoder } from 'util';
 
-export const FusionAuthProvider: React.FC = () => {
-    return <div>Hello</div>;
+export interface IFusionAuthContext {
+    login: (redirectURI: string, state: string) => void;
+}
+
+export const FusionAuthContext = React.createContext<IFusionAuthContext>({
+    login: () => {},
+});
+
+interface Props {
+    baseURL: string;
+    clientID: string;
+    scope: string;
+    children: React.ReactNode;
+}
+
+export const FusionAuthProvider: React.FC<Props> = ({
+    baseURL,
+    clientID,
+    scope,
+    children,
+}) => {
+    const login = useCallback(
+        async (redirectURI: string, state: string) => {
```

Comment:
We should probably pass the redirect URI to the provider to make this call simpler, and also provide a default value for `state` (empty string should be fine). 

Also for casing: for acronyms let's only capitalize if it is two letters. For 3 or more, lowercase the rest of the acronym, so in this case `redirectUri` instead of `redirectURI`